### PR TITLE
Restore Accessibility for Dropdown

### LINKED ISSUE #15172: When selecting a license you can't write in any field
When you try to create a repository and select a license you cannot continue working in the other fields.
I checked the bug in try.gitea.io v1.15.0+dev-27-gdc56fb7c8

### PR DESCRIPTION
**Pull request by zeripath**
This PR restores the vendored and patched dropdown.js from #8638. It
however, it partially abandons the calls to `click()` using instead the default
dropdown click calls instead. This prevents the issue of the dropdown
grabbing focus permanently however, this may have negative effects on
the effect of focus on the dropdowns.

Of note, the behaviour of the template selector dropdown on the repo
creation page is slightly odd - I don't believe that this odd behaviour
is caused by this PR but rather by the feed source for this. I suspect
that the dropdown should be adding a delete button to its selection.

Fix #15172
References: #7057

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Review by techknowlogick — APPROVED** [review]
Tested w/ macOS voiceover and it was able to read the items in the dropdown.

Thanks @zeripath (and also to Jookia for your original work)

**Comment by silverwind** [comment]
Can we at least disable original fomantic module (e.g. remove it from `web_src/fomantic/semantic.json`) so we are not loading 150kB of useless JS again? It probably means you also need to manually insert the dropdown css.

I think a more maintainable way of doing this would be to create a `.patch` file against `web_src/fomantic/build/semantic.js` and apply that during the `make fomantic` step.

**Comment by zeripath** [comment]
> I would suggest to copy the patched file to `web_src/fomantic/node_modules/src/definitions/modules/dropdown.js` during the `make fomantic` target, before the fomantic build runs, that way, we avoid loading the file twice which surely has unintended side-effects.

Done.

> A even better approach would be to create a `.patch` and instead of a full-file replacement, or finally get this patch accepted by upstream.

That doesn't help us right now. The current patch is here:

<details><summary>Patch</summary>

```patch



I think we should be able to just use the hooks within fomantic-ui's dropdown.js to add the

**Comment by silverwind** [comment]
Hmm does that patch cleanly apply on the current `dropdown.js`? If so, I suggest using `git apply` to patch the file on-the fly. I will make the upgrade to fomantic 2.8.8 easier if we can just apply a patch.

**Comment by silverwind** [comment]
Note that the typo fixes that are reverted in this PR are fine, the files should not have been changed like that as they are essentially build artifacts. I patching fails because of these typo changes, revert the files to original build output before applying the patch.

**Comment by Jookia** [comment]
Can you confirm that click handlers work for elements that supply its own click handler, like the contribution permissions dialogs?

File: web_src/fomantic/build/semantic.css [hunk 1]
```diff
@@ -31802,7 +31802,7 @@ Floated Menu / Item
   opacity: 1;
 }
 
-/* Icon Glyph */
+/* Icon Gylph */
```
Comment (lafriks) [inline web_src/fomantic/build/semantic.css:31805]: Why such change?

**Comment by lafriks** [comment]
In a lot of places comments are changed to wrong ones

File: web_src/fomantic/build/semantic.css (hunk 1 above)
Comment (zeripath) [reply web_src/fomantic/build/semantic.css:31805]: These changes come from running make fomantic. 

The spelling changes should never have been made in the first place as they're essentially vendored files.

**Comment by zeripath** [comment]
> In a lot of places comments are changed to wrong ones

These are autogenerated and vendored files. They are the result of `make fomantic`

The spelling changes that are occurring are because the spelling in these files should never have been changed.

**Comment by lafriks** [comment]
Did not notice it's in build, looked like just strange changes in comments from correct to wrong wording 😅

**Comment by kdumontnu** [comment]
Does v1.14 need a backport?

**Comment by fnetX** [comment]
No, v1.14 is unaffected.

**Comment by techknowlogick** [comment]
@lafriks / @a1012112796 you've both reviewed the backport, could you take a look at this PR?

**Review by lafriks — APPROVED** [review]

**Comment by fnetX** [comment]
I waited for it to be deployed to try.gitea.io, and I can confirm that keyboard navigation for dropdowns (e.g. the language selection) is working fine again and that a screen reader reads out the elements (compared to a few minutes before where it didn't), just as with Gitea 1.14.

I have been trying Gitea with a screen reader recently, and I can confirm it's not really optimal. I think I'll open a few issues for stuff I notice, maybe this helps at getting a first idea before asking external auditors, it might also save some money if stuff gets improved before (not sure).

**Comment by fnetX** [comment]
Thank you very much! @zeripath

**Pull request merged**